plugins/main/public/components/endpoints-summary/agent/index.tsx

```diff
@@ -14,81 +10,42 @@ import {
 import { MainAgentStats } from '../../agents/stats';
 import WzManagementConfiguration from '../../../controllers/management/components/management/configuration/configuration-main.js';
-import { getAgentsService } from '../services';
-import { ShareAgent } from '../../../factories/share-agent';
-import { updateCurrentAgentData } from '../../../redux/actions/appStateActions';
 import { endpointSummary } from '../../../utils/applications';
 import { withErrorBoundary, withReduxProvider } from '../../common/hocs';
 import { compose } from 'redux';
+import { PinnedAgentManager } from '../../wz-agent-selector/wz-agent-selector-service';
 
 export const AgentView = compose(
   withErrorBoundary,
   withReduxProvider,
 )(() => {
   //TODO: Replace when implement React router
   const $injector = getAngularModule().$injector;
-  const $router = $injector.get('$route');
   const $commonData = $injector.get('commonData');
 
-  const shareAgent = new ShareAgent();
-
   //TODO: Replace with useDatasource and useSearchBar when replace WzDatePicker with SearchBar in AgentsWelcome component
   const savedTimefilter = $commonData.getTimefilter();
   if (savedTimefilter) {
     getDataPlugin().query.timefilter.timefilter.setTime(savedTimefilter);
     $commonData.removeTimefilter();
   }
 
-  const { agent: agentId } = $router.current.params;
+  const pinnedAgentManager = new PinnedAgentManager();
 
   const [agent, setAgent] = useState<Agent>();
   const [isLoadingAgent, setIsLoadingAgent] = useState(true);
   const [tab, setTab] = useState<string>($commonData.checkTabLocation());
-  const dispatch = useDispatch();
 
   const getAgent = async () => {
-    try {
-      setIsLoadingAgent(true);
-
-      const id = $commonData.checkLocationAgentId(
-        agentId,
-        shareAgent.getAgent(),
-      );
-
-      if (!id) {
-        return;
-      }
-
-      const { affected_items } = await getAgentsService({
-        agents: [agentId],
-      });
-      if (!affected_items?.length) {
-        throw 'Not found';
-      }
-
-      const agent = affected_items[0];
-      setAgent(agent);
-      dispatch(updateCurrentAgentData(agent));
-    } catch (error) {
-      const options = {
-        context: `AgentView.getAgent`,
-        level: UI_LOGGER_LEVELS.ERROR,
-        severity: UI_ERROR_SEVERITIES.CRITICAL,
-        store: true,
-        error: {
-          error,
-          message: error.message || error,
-          title: `Error getting the agent: ${error.message || error}`,
-        },
-      };
-      getErrorOrchestrator().handleError(options);
-    } finally {
-      setIsLoadingAgent(false);
-    }
+    await pinnedAgentManager.syncPinnedAgentSources();
+    const isPinnedAgent = pinnedAgentManager.isPinnedAgent();
+    setAgent(isPinnedAgent ? pinnedAgentManager.getPinnedAgent() : null);
   };
 
   useEffect(() => {
+    setIsLoadingAgent(true);
     getAgent();
+    setIsLoadingAgent(false);
```

Comment:
**nitpick**: the `getAgent` method is async and we are not awaiting the resolution of this and we set the loading agent as `false`. Should we await the resolution to set the loading status?

Reply:
I suggest to put back the lines (setIsLoadingAgent(true) and setIsLoadingAgent(false)) in the function `getAgent`

Reply:
The `setIsLoadingAgent` is moved to `getAgent` so that `isLoadingAgent` is updated correctly when the asynchronous part of `getAgent` finishes